### ISSUE TITLE
$nested.column$ with 1-m and pagination puts where clause on subquery instead of top level

### ISSUE THREAD

**Issue by willbutler** [sequelize/sequelize#11938]
## Issue Description
A subquery is used when eagerly fetching a 1-m association (with pagination). When using the `$nested.column$` syntax in the `where` option, the where clause is put on the subquery instead of the outer query, which results in a `missing FROM-clause entry for table` error.

### What are you doing?
We are trying to use the `$nested.column$` syntax to filter a paginated result set.

**Here is the link to the SSCCE for this issue:** https://github.com/papb/sequelize-sscce/pull/53

### What do you expect to happen?

```
SELECT
   `Foo`.*,
   `Bars`.`id` AS `Bars.id`,
   `Bars`.`name` AS `Bars.name`,
   `Bars`.`FooId` AS `Bars.FooId` 
FROM
   (
      SELECT
         `Foo`.`id`,
         `Foo`.`name` 
      FROM
         `Foos` AS `Foo` LIMIT 10
   )
   AS `Foo` 
   LEFT OUTER JOIN
      `Bars` AS `Bars` 
      ON `Foo`.`id` = `Bars`.`FooId`
      WHERE
         `Bars`.`name` = \'bar\' ;
```

### What is actually happening?

```
SELECT
   `Foo`.*,
   `Bars`.`id` AS `Bars.id`,
   `Bars`.`name` AS `Bars.name`,
   `Bars`.`FooId` AS `Bars.FooId` 
FROM
   (
      SELECT
         `Foo`.`id`,
         `Foo`.`name` 
      FROM
         `Foos` AS `Foo` 
      WHERE
         `Bars`.`name` = \'bar\' LIMIT 10
   )
   AS `Foo` 
   LEFT OUTER JOIN
      `Bars` AS `Bars` 
      ON `Foo`.`id` = `Bars`.`FooId`;
```

```
Error: SQLITE_ERROR: no such column: Bars.name
```

### Environment

- Sequelize version: 5.24.1 
- Node.js version: 10.17.0 
- Operating System: macOS 10.15.3
- If TypeScript related: TypeScript version: XXX

### How does this problem relate to dialects?

- [X] I think this problem happens regardless of the dialect.
- [ ] I think this problem happens only for the following dialect(s): 
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [X] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by mjy78** [comment]
I think if you put `required: true` on the include entry for Bars this should cause the query generator to fetch Bars in the subquery. 

It doesn't suit all scenarios but should work in your case. Where this isn't so helpful is if you're including multiple child models with left outer joins (required:false) and want to craft a predicate that matches at least one of a number of conditions (using OR operator) from fields in the child table.

For example:

```
Parent.findAll({
  include: [ ChildA, ChildB ],  
  where: {
    [Op.or]: [
      { '$ChildA.name': 'Some filter text' },
      { '$ChildB.name': 'Some filter text' }
    ]
  },
  limit: 10, offset: 0
})
```

It would be great if the query generator had some smarts (when using subqueries for pagination or whatever) that detected when a where clause touches a nested model (or deeply nested model) and then automatically included the descendant models in the sub query.

**Comment by bass1387** [comment]
@mjy78 thank you for very useful workaround. Your advice solves `missing FROM-clause entry for table` error for me. It should be noted that adding `required: true` leads to INNER JOIN nested model against LEFT OUTER JOIN with `required: false`.

**Comment by imbrok** [comment]
Considering your expectation, I would try like this.
```js
  Foo.hasMany(Bar);
  Bar.belongsTo(Foo);

  await sequelize.sync();

  log(await Foo.findAndCountAll({include: {model: Bar, where: { name: 'bar' }, required: false}, limit: 10}));
```

I'm not sure this code works for your case.

Cheers.

**Comment by bcolthurst-raices** [comment]
I have use cases where we'd like to do OR conditions in the top-level WHERE clause that requires OUTER joins rather than inner joins.  Happy to help code a solution? Has anyone else run into this problem, or has anyone attempted a solution?

**Comment by bcolthurst-raices** [comment]
@willbutler and @bass1387 
We were running into this issue and we forked the repo intending to code a fix.  We added some tests, got 'em to fail, and got to work. But we soon found that passing in a 
`subQuery: false`
in the top level options fixes the issue, and we're able to do all the left outer joins we want and do top-level nested-column where criteria and the limiting is appropriately "top-level" now, too.  
It feels like a pretty solid work-around.  Curious if anyone else can confirm that it solves it for them, too.
